src/main/java/com/aerotrack/utils/clients/ryanair/RyanairClient.java

```diff
@@ -1,37 +1,53 @@
 package com.aerotrack.utils.clients.ryanair;
 
 import com.aerotrack.model.entities.Flight;
-import com.aerotrack.model.exceptions.ApiRequestException;
-import com.aerotrack.utils.clients.ApiClientWrapper;
 import lombok.AllArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.json.JSONArray;
 import org.json.JSONObject;
+import retrofit2.Retrofit;
+import retrofit2.converter.scalars.ScalarsConverterFactory;
+import retrofit2.http.GET;
+import retrofit2.http.Query;
+import software.amazon.awssdk.services.s3.endpoints.internal.Value;
 
 import java.time.LocalDate;
 import java.time.format.DateTimeFormatter;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Optional;
 
 @Slf4j
 @AllArgsConstructor
 public class RyanairClient {
-    private final ApiClientWrapper apiClientWrapper;
+
+    private final RyanairApiService ryanairApiService;
 
     public static RyanairClient create() {
-        return new RyanairClient(new ApiClientWrapper());
-    }
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl("https://www.ryanair.com/api/booking/v4/it-it/")
+                .addConverterFactory(ScalarsConverterFactory.create())
+                .build();
 
-    private static final String RYANAIR_AVAILABILITY_API = "https://www.ryanair.com/api/booking/v4/it-it/availability?ADT=1&Origin=%s" +
-            "&Destination=%s&IncludeConnectingFlights=false&Disc=0&DateOut=%s&RoundTrip=false&ToUs=AGREED";
+        return new RyanairClient(retrofit.create(RyanairApiService.class));
+    }
 
     public List<Flight> getFlights(String fromAirportCode, String toAirportCode, LocalDate date) {
-        String ryanairURL = String.format(RYANAIR_AVAILABILITY_API, fromAirportCode, toAirportCode, date.format(DateTimeFormatter.ofPattern("yyyy-MM-dd")));
+        try {
+            String formattedDate = date.format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+            String response = ryanairApiService.getFlights("1", fromAirportCode, toAirportCode, "false", "0", formattedDate, "false", "AGREED")
+                    .execute()
+                    .body();
+            return parseFlights(response, fromAirportCode, toAirportCode); // Parse the JSON string
+        } catch (Exception e) {
+            log.error("Error in calling Ryanair API: " + e.getMessage());
+            throw new RuntimeException("Caught exception while calling Ryanair API.", e);
```

Comment:
stai togliendo informazioni utili. Ad esempio se nell'eccezione dice errore 400 io non lo so.
A meno che non vuoi nascondere l'eccezione specifica, ma mi sembra abbastanza importante in questo caso sapere che codice ritorna

Reply:
l'eccezione specifica è nella cause di `e`, però si può fare anche come sopra, faccio un check sul codice ritornato